src/main/kotlin/org/jitsi/jibri/selenium/JibriSelenium.kt

```diff
@@ -226,25 +226,30 @@ class JibriSelenium(
     fun joinCall(callUrlInfo: CallUrlInfo, xmppCredentials: XmppCredentials? = null) {
         // These are all blocking calls, so offload the work to another thread
         TaskPools.ioPool.submit {
-            HomePage(chromeDriver).visit(callUrlInfo.baseUrl)
+            try {
+                HomePage(chromeDriver).visit(callUrlInfo.baseUrl)
 
-            val localStorageValues = mutableMapOf(
-                    "displayname" to jibriSeleniumOptions.displayName,
-                    "email" to jibriSeleniumOptions.email,
-                    "callStatsUserName" to "jibri"
-            )
-            xmppCredentials?.let {
-                localStorageValues["xmpp_username_override"] = "${xmppCredentials.username}@${xmppCredentials.domain}"
-                localStorageValues["xmpp_password_override"] = xmppCredentials.password
-            }
-            setLocalStorageValues(localStorageValues)
-            if (!CallPage(chromeDriver).visit(callUrlInfo.callUrl)) {
+                val localStorageValues = mutableMapOf(
+                        "displayname" to jibriSeleniumOptions.displayName,
+                        "email" to jibriSeleniumOptions.email,
+                        "callStatsUserName" to "jibri"
+                )
+                xmppCredentials?.let {
+                    localStorageValues["xmpp_username_override"] = "${xmppCredentials.username}@${xmppCredentials.domain}"
+                    localStorageValues["xmpp_password_override"] = xmppCredentials.password
+                }
+                setLocalStorageValues(localStorageValues)
+                if (!CallPage(chromeDriver).visit(callUrlInfo.callUrl)) {
+                    stateMachine.transition(SeleniumEvent.FailedToJoinCall)
+                } else {
+                    startRecurringCallStatusChecks()
+                    addParticipantTracker()
+                    currCallUrl = callUrlInfo.callUrl
+                    stateMachine.transition(SeleniumEvent.CallJoined)
+                }
+            } catch (t: Throwable) {
+                logger.error("An error occurred while joining the call", t)
```

Comment:
Did you really want to wrap all of the above instead of targeting the chromedriver-specific lines?